include/sharpSAT/component_types/difference_packed_component.h

```diff
@@ -76,20 +76,20 @@ class DifferencePackedComponent:public BasePackedComponent {
 DifferencePackedComponent::DifferencePackedComponent(Component &rComp) {
 
   unsigned max_var_diff = 0;
-  unsigned hashkey_vars = *rComp.varsBegin();
-  for (auto it = rComp.varsBegin() + 1; *it != varsSENTINEL; it++) {
-    hashkey_vars = (hashkey_vars * 3) + *it;
-    if ((*it - *(it - 1)) - 1 > max_var_diff)
-      max_var_diff = (*it - *(it - 1)) - 1 ;
+  unsigned hashkey_vars = static_cast<unsigned>(*rComp.varsBegin());
+  for (auto it = rComp.varsBegin() + 1; it->var() != varsSENTINEL; it++) {
+    hashkey_vars = (hashkey_vars * 3) + static_cast<unsigned>(*it);
```

Comment:
You should consider introducing a named variable for the underlying value of `*it` and `*(it - 1)`.

Reply:
Just curious... Why would that help?

Reply:
I think it would be more readable than the jumble of explicit static_casts that are there right now.